client/providers/OmnichannelProvider.tsx

```diff
@@ -1,11 +1,15 @@
-import React, { useState, useEffect, FC, useMemo, useCallback, memo } from 'react';
```

Comment:
Do the changes on the provider/context reliant on data already on it? I was thinking maybe we could create a new context/provider for voip, as I feel like these changes will keep on growing, and having it's own context/provider would help down the line when we expand this to the rest of the product. We could add it here https://github.com/RocketChat/Rocket.Chat/blob/livechat-voip/registration-wiring/client/providers/MeteorProvider.tsx#L34

What do you think?

Reply:
@amolghode1981  I've just read the description, did you had issues trying to create it's own provider? What exactly needs to be loaded before the sidebar? Maybe we can find a way to initialize only what is needed on the context (that should load first)

Reply:
@MartinSchoeler, I did create a VoIP provider but the question was where do we place it in MeteorProvider.  We need to have |extensionConfig| and |voipUser| initialised before or with OmnichannelProvider. The reason is that sidebar/sections/Omnichannel.tsx contains a button which needs to have |extensionConfig| and |voipUser|  fully initialised.

As far as understand the working of contexts and providers, a parent gets initialised before the child. So if we create a VoipProvider, it must be a parent of OmnichannelProvider in MeteorProvider hierarchy. This I feel is an inconsistent design. So here we need to find out a solution for loading VoipContext before OmnichannelContext without making a VoipContext, a parent of OmnichannelContext

Reply:
Okay, so we decided to leave the structure as it is for this PR, soon I'll refactor this with the new VoIP Context/provider